jubatus/core/regression/euclidean_distance_regression.cpp

```diff
@@ -31,26 +31,46 @@ namespace core {
 namespace regression {
 
 euclidean_distance_regression::euclidean_distance_regression(
-    size_t k) : inverted_index_regression(k) {
+    const config& conf) : inverted_index_regression(conf) {
 }
 
 float euclidean_distance_regression::estimate(
     const common::sfv_t& fv) const {
   std::vector<std::pair<std::string, float> > ids;
   {
     util::concurrent::scoped_rlock lk(storage_mutex_);
-    mixable_storage_->get_model()->calc_euclid_scores(fv, ids, k_);
+    mixable_storage_->get_model()->calc_euclid_scores(
+        fv, ids, config_.nearest_neighbor_num);
   }
-  float sum = 0.0;
   if (ids.size() > 0) {
-    for (std::vector<std::pair<std::string, float> >::const_iterator
-           it = ids.begin();
-         it != ids.end(); ++it) {
-      const std::pair<bool, uint64_t> index =
-          values_->get_model()->exact_match(it->first);
-      sum += values_->get_model()->get_float_column(0)[index.second];
+    float sum = 0.0;
+    if (config_.weight && *config_.weight == "distance") {
+      float sum_w = 0.0;
+      for (std::vector<std::pair<std::string, float> >:: const_iterator
+               it = ids.begin(); it != ids.end(); ++it) {
+        const std::pair<bool, uint64_t> index =
+            values_->get_model()->exact_match(it->first);
+        float t = values_->get_model()->get_float_column(0)[index.second];
+        // calc_euclid_scores(...) returns minus euclid distance.
+        float d = -1.0 * it->second;
+        if (d == 0.0) {
+          return t;
```

Comment:
> For example, jubatus has (1.0, {"x": 1.0}), (1.1, {"x": 1.0}) and query is {"x": 1.0}.
> In this case, what value should be returned?
I think we should return the MEAN value of the values.
In case of your example, (1.0 + 1.1) / 2 = 1.05 will be returned, do you agree?